controlplane/modules/route/internal/discovery/neigh/neigh.go

```diff
@@ -207,17 +207,19 @@ func (m *NeighMonitor) updateNeighbours() error {
 
 		// Create the entry with resolved hardware addresses
 		entry := NeighbourEntry{
-			NextHop:      nexthopAddr,
-			LinkAddr:     neigh.HardwareAddr,
-			HardwareAddr: hardwareAddr,
-			UpdatedAt:    time.Now(),
-			State:        NeighbourState(neigh.State),
+			NextHop: nexthopAddr,
+			HardwareRoute: HardwareRoute{
+				SourceMAC:      [6]byte(hardwareAddr),
+				DestinationMAC: [6]byte(neigh.HardwareAddr),
+			},
+			UpdatedAt: time.Now(),
+			State:     NeighbourState(neigh.State),
 		}
 
 		m.log.Debugw("resolved neighbour entry",
 			zap.Stringer("nexthop_addr", entry.NextHop),
-			zap.Stringer("nexthop_hardware_addr", entry.LinkAddr),
-			zap.Stringer("hardware_addr", entry.HardwareAddr),
+			zap.Stringer("nexthop_hardware_addr", neigh.HardwareAddr),
+			zap.Stringer("inface_hardware_addr", hardwareAddr),
```

Comment:
What was wrong with previous attribute?

Reply:
There is too much hardware around; we need some way to distinguish between them. :)